### PR TITLE
fix: inject hangs with undefined promise resolve

### LINKED ISSUE #4207: "inject" doesn't respond when route doesn't have a return statement
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.5.2

### Plugin version

_No response_

### Node.js version

16.16

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

5.15.60-1-MANJARO

### Description

We've captured this issue while testing our routes which don't have a `return` statement:

- Browser behavior is normal when accessing the route;
- `inject` method hangs while awaiting a response from the route;

### Steps to Reproduce

Create a Fastify server with a `GET` route to "/" with no `return` statement.

```
function buildServer() {
  const fastify = require('fastify')()
  fastify.get('/', async function (request, reply) {})
  return fastify
}
```

- Build the server and notice how the route works in the Browser;

```
const server = buildServer()
server.listen({ port: 3000 })
```

- Invoke `inject` for the route;

```
async function testInject() {
  const response = await server.inject({
    method: 'GET',
    url: '/'
  })

  console.log("expecting a response", response);
}
```

Notice how `console.log` is not reached, and the server hangs.

### Expected Behavior

If we omit `return` (or `reply.send({})`) the Browser still responds. But `inject` hangs.
We'd like to `inject` to have the same behavior as the browser, then we would be able to create async routes without `return` statements and/or non-async routes with no `reply.send({})` statements.

### PR DESCRIPTION
**Pull request by simoneb**
Closes #4207 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/async-await.test.js [hunk 1]
```diff
@@ -423,6 +423,28 @@ test('promise was fulfilled with undefined', t => {
...
+  fastify.get('/', async (req, reply) => {
+  })
+
+  stream.once('data', line => {
+    t.fail('should not log an error')
+  })
+
+  const res = await fastify.inject('/')
+
+  t.equal(res.body, '')
```
Comment (simoneb) [inline test/async-await.test.js:444]: the body should be null, but I guess it's another edge case due to mimicking built-in Request/Response in light-my-request

Comment (mcollina) [reply test/async-await.test.js:444]: There is no such thing as a null http body, it's either a string or buffer of lenght 0.

**Comment by mcollina** [comment]
Can you update the PR title?

File: test/async-await.test.js (hunk 1 above)
Comment (simoneb) [reply test/async-await.test.js:444]: well if you look at the test in the same file with a plan http client, you can see that it asserts against undefined https://github.com/fastify/fastify/pull/4211/files#diff-1c26a7df32f7d09a699227287432ba17f6a97376fd6e3e6f334256b2f5b4c526R420

**Comment by simoneb** [comment]
> Can you update the PR title?

done!

File: test/async-await.test.js (hunk 1 above)
Comment (mcollina) [reply test/async-await.test.js:444]: That's how sget is implementing it, I would argue it's not really correct.

**Review by mcollina — APPROVED** [review]
lgtm
